applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/prebuckling_strategy.hpp

```diff
@@ -0,0 +1,980 @@
+    {
+        KRATOS_TRY
+
+        ModelPart &rModelPart = BaseType::GetModelPart();
+        const int rank = rModelPart.GetCommunicator().MyPID();
+
+        KRATOS_INFO_IF("PrebucklingStrategy", BaseType::GetEchoLevel() > 2 && rank == 0)
+            << "Entering Check" << std::endl;
+
+        // check the model part
+        BaseType::Check();
+
+        // check the scheme
+        this->pGetScheme()->Check(rModelPart);
+
+        // check the builder and solver
+        this->pGetBuilderAndSolver()->Check(rModelPart);
+
+        KRATOS_INFO_IF("PrebucklingStrategy", BaseType::GetEchoLevel() > 2 && rank == 0)
+            << "Exiting Check" << std::endl;
+
+        return 0;
+
+        KRATOS_CATCH("")
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    SchemePointerType mpScheme;
+
+    BuilderAndSolverPointerType mpEigenSolver;
+
+    BuilderAndSolverPointerType mpBuilderAndSolver;
+
+    // SparseMatrixPointerType mpMassMatrix;
+
+    SparseMatrixPointerType mpStiffnessMatrix;
+
+    SparseMatrixPointerType mpStiffnessMatrixPrevious;
+
+    SparseVectorPointerType mpRHS;
+
+    SparseVectorPointerType mpDx;
+
+    typename ConvergenceCriteriaType::Pointer mpConvergenceCriteria;
+    /// The pointer to the convergence criteria employed
+
+    bool mInitializeWasPerformed = false;
+
+    bool mSolutionStepIsInitialized = false;
+
+    bool mSolutionFound = false;
+
+    unsigned int mLoadStepIteration = 0;
+
+    unsigned int mMaxIteration;
+
+    std::vector<array_1d<double,3>> mpInitialLoads;
+
+    double mInitialStep;
+    double mSmallStep;
+    double mBigStep;
+    double mConvergenceRatio;
+
+    double lambda = 0.0;
+    double lambdaPrev = 1.0;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+/*
+Store Initial Loads to ensure that load multiplier always refers to user input
+*/
+    void StoreInitialLoadConditions()
+    {
+        ModelPart &rModelPart = BaseType::GetModelPart();
+        const auto it_conditions_begin = rModelPart.ConditionsBegin();
+        for (int j = 0; j < static_cast<int>(rModelPart.Conditions().size()); j++) 
+        {
+            auto it_condition = it_conditions_begin + j;
+            // Check if condition is point load
+            if (it_condition->Has(POINT_LOAD))
```

Comment:
Here only `POINT_LOAD` and `SURFACE_LOAD` are considered. @philbucher i am pretty sure there are other loads as well. Like this only certain loads will be considered to be dependent on the load factor.

 I think this is related to @loumalouomega comment:

> I am against this, I think is more intelligent to define a Variable like: buckling coefficient. (BUCKLING_COEFFICIENT) and define as a property, then check this variable in all load conditions, otherwise take 1. Therefore you will need to change the variable only once. Additionally some people apply loads in the nodes, so this will not work

Reply:
Okay, so then we do it like that and I change the implementation of all conditions?

Reply:
Before you start doing that let's wait for @philbucher or other opinions because thats something that should be discussed first.

Reply:
to me modifying all conditions for this is a no-go

Reply:
Any more suggestions how we wanna proceed?